### PR TITLE
Fix multiple issues with streams and redirection

### LINKED ISSUE #6642: Various corrections and improvements for about_Redirection
Re section "Long description":

> By default, PowerShell sends its command output to the PowerShell _console_.

Since not all hosts are consoles, it's better to say something like "By default, PowerShell sends its command output to the host, which in a console (terminal) window means printing to the screen."

> Use the PowerShell redirection operators.

While it can be _inferred_ from later statements, it would help to point out that the redirection operators with a file target are functionally equivalent to piping to `Out-File` calls without additional arguments (except `-Append` in the case of `>>`).

---

Re section "PowerShell redirection operators"

While it is perhaps easy to guess, it would still be helpful to add a column to the table that names the stream-specific `Write-*` cmdlets and links to their topics.

---

Re section "Notes":

> When you are writing to files, the redirection operators use Unicode encoding. If the file has a different encoding, the output might not be formatted correctly. To redirect content to non-Unicode files, use the Out-File cmdlet with its Encoding parameter.

This only applies to _Windows PowerShell_, where `Unicode` encoding - i.e. UTF-16LE - is the default (which differs from `Set-Content` there, for instance).

In PowerShell [Core] v6+, BOM-less UTF-8 is now the default, consistently applied across across all cmdlets.

---

Typo:

> Show the affect of Action Preferences

"affect" -> "effect"

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c276a962-ace4-a158-8b48-5181e58a3def
* Version Independent ID: 2d6a2602-4480-d4d2-d63d-739c4d93e4b1
* Content: [about_Redirection - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_redirection?view=powershell-7)
* Content Source: [reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md)
* Product: **powershell**
* Technology: **powershell-conceptual**
* GitHub Login: @SteveL-MSFT
* Microsoft Alias: **slee**

### LINKED ISSUE #6643: Introduce a conceptual help topic about PowerShell's output streams: about_Streams
# Summary of the new document or enhancement

* As a user, I want to understand the respective purpose of PowerShell's 6 output streams, and what content I should expect in each.

* As a developer, I want guidance on what information should be sent to what stream.

A new conceptual `about_Streams` topic could cover these topics (and would _complement_ `about_Redirection`, which is only about _where to send_ stream output).

---

As for guidance, @KirkMunro had some thoughts in https://github.com/PowerShell/PowerShell/issues/13636#issuecomment-693429786:

> I also think PowerShell streams are misunderstood, and therefore misused. My take:
> *  Verbose is for messages that will help users troubleshoot why a command that they invoked interactively or in their script may have failed, when the error message itself is not enough. It is also to report information that is not otherwise available (site redirects on a REST API invocation, etc.), which again is intended to help someone understand a bit more technical detail to troubleshoot issues.

> *  Debug is for messages that will help a command author understand why their command is failing when someone else invoked it (or when they invoked it if they are actively debugging it). These can contain more internal details, and there may be many debug messages if they are necessary for deep troubleshooting, but their target audience is command authors

> *  Progress is for messages that communicate progress in longer running commands or scripts.

> *  Information is for messages that help a user understand what is being done in a script (if they are reviewing an ongoing run log or console where a command is running) or what was done by a script (if they are reviewing output of a script after the fact). It is also for developers to have additional streams if they need to pass information through PowerShell, where they can tag the stream and then have specific handling for that stream, passing object data out to a runtime.

Details of requested document:

- Proposed title: `about_Streams`
- Propose location in the TOC:
- Target audience: end users and developers
- List of related articles to link to: `about_Redirection`, various `Write-*` cmdlets

### LINKED ISSUE #6644: Confusion between verbose and debug streams
(Emphasis added.)

> Typically, the **verbose** message stream is used to deliver information about command processing that is **used for debugging a command**.

Given that there's a separate _debug_ stream with a separate cmdlet, `Write-Debug`, it is confusing to refer to debugging here.

Instead, a brief reference to the purpose of verbose output - as distinct from debug output - would be helpful.

If and when #6643 gets implemented, we could link there for more detailed information.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9ccff74d-f790-2824-58fe-007084ffa5a2
* Version Independent ID: 32088f22-8706-87ca-9e08-124379088e50
* Content: [Write-Verbose (Microsoft.PowerShell.Utility) - PowerShell](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/write-verbose?view=powershell-7.1&WT.mc_id=ps-gethelp)
* Content Source: [reference/7.1/Microsoft.PowerShell.Utility/Write-Verbose.md](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.1/Microsoft.PowerShell.Utility/Write-Verbose.md)
* Product: **powershell**
* Technology: **powershell-cmdlets**
* GitHub Login: @JamesWTruher
* Microsoft Alias: **jimtru**

### PR DESCRIPTION
**Pull request by sdwheeler**
# PR Summary

Fixes [AB#1782271](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1782271) - Fixes #6642 - update redirection info
Fixes [AB#1782991](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1782991) - Fixes #6643 - create about_Output_Streams
Fixes [AB#1782991](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1782991) - Fixes #6644 - clarify purpose of Verbose and Debug streams

## PR Context

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [ ] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Version 7.1 preview content
- [x] Version 7.0 content
- [x] Version 6 content
- [x] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+Debug messages are intended for script and cmdlet developers more than end
+users. These debug messages can contain internal details necessary for deep
+troubleshooting.
+
+## Information stream
+
+The **Information** stream is intended to provide message that help a user
+understand what a script is doing. It can also be used by developers as an
+additional streams used to pass information through PowerShell. The developer
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:87]: streams -> stream.  And is it really "passing information through PowerShell"?  I'd rephrase that.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+can tag stream data and have specific handling for that stream. Use the
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:88]: Link to how to tag stream data maybe?

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+`Write-Information` cmdlet to explicitly write to this stream.
+
+## Progress stream
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:91]: Since this gets an H2, I'm pretty sure the above table should include progress in table instead of as a note.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 4]
```diff
@@ -27,30 +27,30 @@ You can use the following methods to redirect output:
...
-| *        | All Streams            | PowerShell 3.0 |
+| Stream # |      Description       | Introduced in  |    Write Cmdlet     |
+| -------- | ---------------------- | -------------- | ------------------- |
+| 1        | **Success** Stream     | PowerShell 2.0 | `Write-Output`      |
+| 2        | **Error** Stream       | PowerShell 2.0 | `Write-Error`       |
+| 3        | **Warning** Stream     | PowerShell 3.0 | `Write-Warning`     |
+| 4        | **Verbose** Stream     | PowerShell 3.0 | `Write-Verbose`     |
+| 5        | **Debug** Stream       | PowerShell 3.0 | `Write-Debug`       |
+| 6        | **Information** Stream | PowerShell 5.0 | `Write-Information` |
+| *        | All Streams            | PowerShell 3.0 |                     |
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:46]: Not sure the point of this entry?

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 5]
```diff
@@ -27,30 +27,30 @@ You can use the following methods to redirect output:
...
 
 > [!NOTE]
-> There is also a **Progress** stream in PowerShell, but it is not used for
+> There is also a **Progress** stream in PowerShell, but it is not support
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:49]: Include in table above, imho.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 6]
```diff
@@ -202,17 +202,17 @@ Inquire
...
 
 To force the redirection of content to a read-only, hidden, or system file,
 use the `Out-File` cmdlet with its `Force` parameter.
 
-When you are writing to files, the redirection operators use Unicode encoding.
-If the file has a different encoding, the output might not be formatted
-correctly. To redirect content to non-Unicode files, use the `Out-File` cmdlet
-with its `Encoding` parameter.
+When you are writing to files, the redirection operators use UTF8 encoding
+(without a BOM). If the file has a different encoding, the output might not be
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:213]: Should BOM get spelled out?  I'm not sure everyone will understand this pseudo-comment.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md [hunk 7]
```diff
@@ -202,17 +202,17 @@ Inquire
...
+formatted correctly. To write to files with a different encoding, use the
+`Out-File` cmdlet with its `Encoding` parameter.
```
Comment (DCtheGeek) [inline reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:215]: The encoding parameter should be bold, not code fence.  This article looks like it needs a style pass, too.

**Review by DCtheGeek — CHANGES_REQUESTED** [review]

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+| 1        | **Success** stream     | PowerShell 2.0 | `Write-Output`      |
+| 2        | **Error** stream       | PowerShell 2.0 | `Write-Error`       |
+| 3        | **Warning** stream     | PowerShell 3.0 | `Write-Warning`     |
+| 4        | **Verbose** stream     | PowerShell 3.0 | `Write-Verbose`     |
+| 5        | **Debug** stream       | PowerShell 3.0 | `Write-Debug`       |
+| 6        | **Information** stream | PowerShell 5.0 | `Write-Information` |
+
+> [!NOTE]
+> There is also a **Progress** stream in PowerShell. Use `Write-Progress`
+> cmdlet to write to this stream. This stream does not support redirection.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:35]: Edited

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+title: about_Output_Streams
+---
+# About output streams
+
+## Short description
+Explains the availability and purpose of output streams in PowerShell.
+
+## Long description
+
+PowerShell provides multiple output streams. The streams provide different
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:17]: removed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+| 6        | **Information** stream | PowerShell 5.0 | `Write-Information` |
+
+> [!NOTE]
+> There is also a **Progress** stream in PowerShell. Use `Write-Progress`
+> cmdlet to write to this stream. This stream does not support redirection.
+
+## Success stream
+
+The **Success** stream is the default stream for normal, successful results.
+Use the `Write-Output` cmdlet to explicitly write objects to this stream. This
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:40]: fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+is the stream used for transporting objects through the PowerShell pipeline.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:41]: fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+The **Success** stream is connected to the **stdout** stream for native
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:42]: stet

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+applications.
+
+## Error stream
+
+The **Error** stream is the default stream for error results. Use the
+`Write-Error` cmdlet to explicitly write to this stream. The **Error** stream
+is connected to the **stderr** stream for native applications. Under most
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:49]: stet

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+The **Warning** stream is intended for error conditions that are less severe
+than errors written to the **Error** stream. Under normal conditions, these
+warnings do not terminate execution. Warnings are not written to the `$Error`
+automatic variable. Use the `Write-Warning` cmdlet to explicitly write to this
+stream.
+
+## Verbose stream
+
+The **Verbose** stream is intended for messages that help users troubleshoot
+command as they are run interactively or from a script. Use the `Write-Verbose`
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:65]: fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+cmdlet to explicitly write messages to this stream. Many cmdlets provide
+verbose output that useful for understanding internal works of the cmdlet. The
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:67]: fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+verbose messages are output only when you use the `-Verbose` common parameter.
+For more information, see [about_CommonParameters](about_CommonParameters.md).
+
+## Debug stream
+
+The **Debug** stream is for messages that scripters understand why their
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:73]: Fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md (hunk 1 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:87]: Fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md (hunk 2 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:88]: No good example to link to.

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md (hunk 3 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Output_Streams.md:91]: Fixed

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 5 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:49]: fixed grammar

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 6 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:213]: stet

File: reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md (hunk 7 above)
Comment (sdwheeler) [reply reference/7.0/Microsoft.PowerShell.Core/About/about_Redirection.md:215]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md [hunk 17]
```diff
@@ -361,6 +361,10 @@ a terminating error condition. To signal a terminating error, use the
 
 [about_Automatic_Variables](../microsoft.powershell.core/about/about_automatic_variables.md)
 
+[about_Output_Streams](/powershell/module/Microsoft.PowerShell.Core/About/about_Output_Streams.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md:364]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md [hunk 18]
```diff
@@ -361,6 +361,10 @@ a terminating error condition. To signal a terminating error, use the
...
+
+[about_Redirection](/powershell/module/Microsoft.PowerShell.Core/About/about_Redirection.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md:366]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md [hunk 19]
```diff
@@ -166,6 +182,10 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
 
 ## RELATED LINKS
 
+[about_Output_Streams](/powershell/module/Microsoft.PowerShell.Core/About/about_Output_Streams.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md:185]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md [hunk 20]
```diff
@@ -166,6 +182,10 @@ This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable
...
+
+[about_Redirection](/powershell/module/Microsoft.PowerShell.Core/About/about_Redirection.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md:187]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md [hunk 21]
```diff
@@ -150,6 +150,10 @@ You can pipe objects to `Write-Output`.
 
 ## RELATED LINKS
 
+[about_Output_Streams](/powershell/module/Microsoft.PowerShell.Core/About/about_Output_Streams.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md:153]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md [hunk 22]
```diff
@@ -150,6 +150,10 @@ You can pipe objects to `Write-Output`.
...
+
+[about_Redirection](/powershell/module/Microsoft.PowerShell.Core/About/about_Redirection.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md:155]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md [hunk 23]
```diff
@@ -98,10 +101,15 @@ You can pipe a string that contains the message to `Write-Verbose`.
   more information, see [about_CommonParameters](https://go.microsoft.com/fwlink/?LinkID=113216).
 - In Windows PowerShell background jobs and remote commands, the `$VerbosePreference` variable in
   the job session and remote session determine whether the verbose message is displayed by default.
-  For more information about the `$VerbosePreference` variable, see [about_Preference_Variables](../Microsoft.PowerShell.Core/About/about_Preference_Variables.md).
+  For more information about the `$VerbosePreference` variable, see
+  [about_Preference_Variables](../Microsoft.PowerShell.Core/About/about_Preference_Variables.md).
 
 ## RELATED LINKS
 
+[about_Output_Streams](/powershell/module/Microsoft.PowerShell.Core/About/about_Output_Streams.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md:109]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md [hunk 24]
```diff
@@ -98,10 +101,15 @@ You can pipe a string that contains the message to `Write-Verbose`.
...
+
+[about_Redirection](/powershell/module/Microsoft.PowerShell.Core/About/about_Redirection.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md:111]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 25]
```diff
@@ -21,58 +21,69 @@ Write-Warning [-Message] <String> [<CommonParameters>]
 ```
 
 ## DESCRIPTION
-The **Write-Warning** cmdlet writes a warning message to the Windows PowerShell host.
-The response to the warning depends on the value of the user's $WarningPreference variable and the use of the *WarningAction* common parameter.
+
+The `Write-Warning` cmdlet writes a warning message to the PowerShell host. The response to the
+warning depends on the value of the user's $WarningPreference variable and the use of the
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:26]: Should the variable be in a code fence?

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 26]
```diff
@@ -21,58 +21,69 @@ Write-Warning [-Message] <String> [<CommonParameters>]
...
 ### Example 3: Set the $WarningPreference variable and write a warning
 
-```
-PS C:\> $warningpreference
-Continue PS C:\> Write-Warning "This is only a test warning."
-This is only a test warning. PS C:\> $warningpreference = "SilentlyContinue"
+This example shows the effect of the value of the `$WarningPreference` variable on a `Write-Warning`
+command.
+
+```powershell
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:55]: While this is being updated, should we cleanup the PS prompt?

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 27]
```diff
@@ -21,58 +21,69 @@ Write-Warning [-Message] <String> [<CommonParameters>]
...
+PS C:\> $WarningPreference
+Continue
+PS C:\> Write-Warning "This is only a test warning."
+This is only a test warning. PS C:\> $WarningPreference = "SilentlyContinue"
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:59]: Don't think these should be 1 line.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 28]
```diff
@@ -21,58 +21,69 @@ Write-Warning [-Message] <String> [<CommonParameters>]
...
 At line:1 char:14
      + Write-Warning <<<<  "This is only a test message."
 ```
 
-This example shows the effect of the value of the $WarningPreference variable on a **Write-Warning** command.
-
-The first command displays the default value of the $WarningPreference variable, which is Continue.
-As a result, when you write a warning, the warning message is displayed and execution continues.
+The first command displays the default value of the `$WarningPreference` variable, which is
+Continue. As a result, when you write a warning, the warning message is displayed and execution
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:70]: Should `Continue` be formatted since it's a value?

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 29]
```diff
@@ -21,58 +21,69 @@ Write-Warning [-Message] <String> [<CommonParameters>]
...
+continues.
 
-When you change the value of the $WarningPreference variable, the effect of the **Write-Warning** command changes again.
-A value of SilentlyContinue suppresses the warning.
-A value of Stop displays the warning and then stops execution of the command.
+When you change the value of the `$WarningPreference` variable, the effect of the `Write-Warning`
+command changes again. A value of `SilentlyContinue` suppresses the warning. A value of Stop
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:74]: Same with `Stop`?

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 30]
```diff
@@ -108,27 +117,37 @@ Accept wildcard characters: False
...
 
-*
+The default value for the `$WarningPreference` variable is `Continue`, which displays the warning
+and then continues executing the command. To determine valid values for a preference variable such
+as `$WarningPreference`, set it to a string of random characters, such as "abc". The resulting error
+message lists the valid values.
 
 ## RELATED LINKS
 
+[about_Output_Streams](/powershell/module/Microsoft.PowerShell.Core/About/about_Output_Streams.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:147]: **Site Relative** links shouldn't include the extension.

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 31]
```diff
@@ -108,27 +117,37 @@ Accept wildcard characters: False
...
+
+[about_Redirection](/powershell/module/Microsoft.PowerShell.Core/About/about_Redirection.md)
```
Comment (DCtheGeek) [inline reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:149]: **Site Relative** links shouldn't include the extension.

**Review by DCtheGeek — CHANGES_REQUESTED** [review]

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md [hunk 32]
```diff
@@ -129,12 +133,17 @@ You can pipe a string that contains a debug message to `Write-Debug`.
 
 ### None
 
-`Write-Debug` only writes to the debug stream. It does not return any output.
+`Write-Debug` only writes to the debug stream. It does not any objects to the
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md:136]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md [hunk 33]
```diff
@@ -129,12 +133,17 @@ You can pipe a string that contains a debug message to `Write-Debug`.
...
+pipeline.
 
 ## NOTES
 
 ## RELATED LINKS
 
+[about_Output_Streams](/powershell/module/Microsoft.PowerShell.Core/About/about_Output_Streams.md)
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md:143]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md [hunk 34]
```diff
@@ -129,12 +133,17 @@ You can pipe a string that contains a debug message to `Write-Debug`.
...
+
+[about_Redirection](/powershell/module/Microsoft.PowerShell.Core/About/about_Redirection.md)
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Debug.md:145]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md (hunk 17 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md:364]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md (hunk 18 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Error.md:366]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md (hunk 19 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md:185]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md (hunk 20 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Information.md:187]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md (hunk 21 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md:153]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md (hunk 22 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Output.md:155]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md (hunk 23 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md:109]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md (hunk 24 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Verbose.md:111]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 25 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:26]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 26 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:55]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 27 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:59]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 28 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:70]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 29 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:74]: Fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 30 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:147]: fixed

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md (hunk 31 above)
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md:149]: fixed

**Review by DCtheGeek — CHANGES_REQUESTED** [review]

File: reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md [hunk 35]
```diff
@@ -21,74 +21,84 @@ Write-Warning [-Message] <String> [<CommonParameters>]
...
+[about_Preference_Variables](../Microsoft.Powershell.Core/About/about_Preference_Variables).
 
 ### Example 4: Set the WarningAction parameter and write a warning
 
-```
-PS C:\> Write-Warning "This is only a test warning." -WarningAction Inquire
+This example shows the effect of the **WarningAction** common parameter on a `Write-Warning`
+command. You can use the **WarningAction** common parameter with any cmdlet to determine how
+PowerShell responds to warnings resulting from that command. The **WarningAction** common parameter
+overrides the value of the $WarningPreference only for that particular command.
```
[the remark this answers is not in the document]
Comment (sdwheeler) [reply reference/5.1/Microsoft.PowerShell.Utility/Write-Warning.md]: fixed
